### PR TITLE
Set CMAKE_MODULE_PATH appropriately to always use preferred linalg-modules

### PR DESCRIPTION
**Pull request by ajaypanyala**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,8 +6,8 @@ set( FETCHCONTENT_UPDATES_DISCONNECTED ON CACHE BOOL "Disable FC Updates" )
 project( GauXC VERSION 0.0.1 LANGUAGES C CXX )
 
 # Place local modules in the path
-list( APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake )
-list( APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules )
+list(INSERT CMAKE_MODULE_PATH 0 ${PROJECT_SOURCE_DIR}/cmake )
```
Comment (wavefunction91) [inline CMakeLists.txt:9]: Isn't this equivalent to `list( PREPEND`?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -45,10 +48,14 @@ configure_file(
...
+if(GAUXC_ENABLE_OPENMP)
+  set(gauxc_omp_target OpenMP::OpenMP_CXX)
+endif()
+
 target_link_libraries( gauxc PUBLIC 
   ExchCXX::ExchCXX 
   IntegratorXX::IntegratorXX 
-  OpenMP::OpenMP_CXX
+  ${gauxc_omp_target}
 )
```
Comment (wavefunction91) [inline src/CMakeLists.txt:59]: I would change this to
```
if( GAUXC_ENABLE_OPENMP )
  target_link_libraries( gauxc PUBLIC OpenMP::OpenMP_CXX )
endif()
```

**Review by wavefunction91 — CHANGES_REQUESTED** [review]

**Comment by wavefunction91** [comment]
@ajaypanyala Is there anything else you want to add to this? I'm going to hold off on merging until https://github.com/wavefunction91/linalg-cmake-modules/pull/1 is merged

**Comment by ajaypanyala** [comment]
I am done with my edits. Thanks!

File: cmake/gauxc-dep-versions.cmake [hunk 3]
```diff
@@ -1,5 +1,5 @@
-set( GAUXC_LINALG_MODULES_REPOSITORY https://github.com/wavefunction91/linalg-cmake-modules.git )
-set( GAUXC_LINALG_MODULES_REVISION  89c82b78505f60657ef33b5bf38e5da059dff7f2 ) 
+set( GAUXC_LINALG_MODULES_REPOSITORY https://github.com/ajaypanyala/linalg-cmake-modules )
```
Comment (ajaypanyala) [inline cmake/gauxc-dep-versions.cmake:1]: Note before merging: This needs to change once [wavefunction91/linalg-cmake-modules#1](https://github.com/wavefunction91/linalg-cmake-modules/pull/1) is merged

File: cmake/gauxc-dep-versions.cmake [hunk 4]
```diff
@@ -1,5 +1,5 @@
...
+set( GAUXC_LINALG_MODULES_REVISION  main )
```
Comment (ajaypanyala) [inline cmake/gauxc-dep-versions.cmake:2]: TODO: Pin commit before merging.

**Pull request closed without merging**